src/lang/std/stdTypes.ts

```diff
@@ -141,17 +144,11 @@ export type RawArgs = Array<RawArg>
  * when we only need to know where there arg is.
  */
 export type SimplifiedArgDetails =
-  | {
-      type: 'singleValue'
-    }
-  | { type: 'arrayItem'; index: 0 | 1 }
-  | { type: 'objectProperty'; key: InputArgKeys }
-  | {
-      type: 'arrayInObject'
-      key: InputArgKeys
-      index: 0 | 1
-    }
-
+  | Omit<SingleValueInput<null>, 'expr' | 'argType'>
+  | Omit<ArrayItemInput<null>, 'expr' | 'argType'>
+  | Omit<ObjectPropertyInput<null>, 'expr' | 'argType'>
+  | Omit<ArrayOrObjItemInput<null>, 'expr' | 'argType'>
+  | Omit<ArrayInObject<null>, 'expr' | 'argType'>
```

Comment:
Conditional types are distributive over union types. You could write the following to prevent accidentally forgetting an Input type:

```suggestion
export type SimplifiedArgDetails = Omit<_InputArg<null>, 'expr' | 'argType'>
```

Pretty sick right?

Reply:
I tried this at first, and the type you get from this is definitely not right
![image](https://github.com/user-attachments/assets/e0cb1215-4177-4f00-af23-f3ec26bcb35e)
I mess around with some more advanced types that looped of each of the unions in `_InputArg` but all of it was much hard to read so this is what I needed up with.

I don't know why `Omit<_InputArg<null>, 'expr' | 'argType'>` it seems intuitive to me that that's how it should work but 🤷